tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -0,0 +1,217 @@
+    info 'Reading object list from S3...'
+    bucket = Aws::S3::Bucket.new(DEFAULT_S3_BUCKET)
+    bucket.objects.each do |object_summary|
+      animation_name = object_summary.key[/^[^.]+/]
+      extension = object_summary.key[/(?<=\.)\w+$/]
+      verbose <<-EOS.unindent
+        #{bold object_summary.key}
+          #{object_summary.last_modified} | #{object_summary.size}
+      EOS
+      # Push into animations collection
+      animations_by_name[animation_name] ||= {}
+      animations_by_name[animation_name][extension] = object_summary
+    end
+
+    # TODO: Validate that keys are unique
+
+    info "Found #{animations_by_name.size} animations."
+
+    animation_metadata_by_name = {}
+    info "Building animation metadata..."
+    animations_by_name.each do |name, objects|
+      # TODO: Validate that every JSON is paired with a PNG and vice-versa
+      # Actually download the JSON from S3
+      json_response = objects['json'].get
+      metadata = JSON.parse(json_response.body.read)
+      # TODO: Validate metadata, ensure it has everything it needs
+      # Record target version in the metadata, so environments (and projects)
+      # consistently reference the version they originally imported.
+      metadata['version'] = objects['png'].object.version_id
+
+      # Generate appropriate sourceUrl pointing to the animation library API
+      metadata['sourceUrl'] = "/api/v1/animation-library/#{metadata['version']}/#{name}.png"
+
+      # Populate sourceSize if not already present
+      unless metadata.key?('sourceSize')
+        png_body = objects['png'].object.get.body.read
+        metadata['sourceSize'] = dimensions_from_png(png_body)
+      end
+
+      animation_metadata_by_name[name] = metadata
+      verbose <<-EOS
+#{bold name} @ #{metadata['version']}
+#{JSON.pretty_generate metadata}
+      EOS
+    end
+
+    info "Metadata built for #{animation_metadata_by_name.size} animations."
+
+    info "Building alias map..."
+
+    alias_map = {}
+    animation_metadata_by_name.each do |name, metadata|
+      aliases = [name] + (metadata['aliases'] || [])
+      aliases.each do |aliaz|
+        # Push name into target array, deduplicate, and sort
+        alias_map[aliaz] = ((alias_map[aliaz] || []) + [name]).uniq.sort
+      end
+    end
+
+    alias_map.each {|k, v| verbose "#{bold k}: #{v.join(', ')}"} if @options[:verbose]
+
+    info "Mapped #{alias_map.size} aliases."
+
+    # Write result to file
+    File.open(DEFAULT_OUTPUT_FILE, 'w') do |file|
+      file.write(JSON.pretty_generate({
+          'metadata': animation_metadata_by_name,
+          'aliases': alias_map
+      }))
+    end
+    info <<-EOS.unindent
+      Manifest written to #{DEFAULT_OUTPUT_FILE}.
+
+        #{dim 'd[ o_0 ]b'}
+    EOS
+
+  # Report any issues while talking to S3 and suggest most likely steps for fixing it.
+  rescue Aws::Errors::ServiceError => service_error
```

Comment:
This catches any S3-related error, but the most common cause will be missing credentials so I make sure to print a hint about that if it occurs.